samcli/commands/local/cli_common/options.py

```diff
@@ -121,6 +134,9 @@ def construct_options(f):
             click.option(
                 "--port", "-p", default=port, help="Local port number to listen on (default: '{}')".format(str(port))
             ),
+            click.option("--ssl-cert-file", default=None, callback=validate_ssl_params, help="Path to SSL certificate file (default: None)"),
+            # Mark --ssl-key-file as eager, so by the time the --ssl-cert-file validator is invoked, we know if it's missing
+            click.option("--ssl-key-file", default=None, callback=validate_ssl_params, is_eager=True, help="Path to SSL key file (default: None)"),
```

Comment:
Sorry that I missed this in the first round but we could use Click's file path arguments which will provide some basic path validation out of the box. https://click.palletsprojects.com/en/8.1.x/arguments/#file-path-arguments

Reply:
Nice catch, thanks. That does make things a lot simpler, and eliminates the need to call the custom validator on the first pass.